ddl/backfilling_operator.go

```diff
@@ -141,6 +147,60 @@ func NewAddIndexIngestPipeline(
 	), nil
 }
 
+// NewWriteIndexToExternalStoragePipeline creates a pipeline for writing index to external storage.
+func NewWriteIndexToExternalStoragePipeline(
+	ctx *OperatorCtx,
+	store kv.Storage,
+	sessPool opSessPool,
+	sessCtx sessionctx.Context,
+	jobID int64,
+	subtaskID int64,
+	tbl table.PhysicalTable,
+	idxInfo *model.IndexInfo,
+	startKey, endKey kv.Key,
+	totalRowCount *atomic.Int64,
+	onClose external.OnCloseFunc,
```

Comment:
11 input parameters, maybe use a struct to wrap them?

Reply:
Do you mean `type XXXArgs struct { ... }`? Then the constructor of `XXXArgs` also has 11 parameters...

Reply:
ok, that's ugly too 😅

Reply:
11 fields seems more widely used than 11 arguments. like https://pkg.go.dev/github.com/cockroachdb/pebble#Options

Reply:
in that case, most options are actually optional